cmd/ocm/list/org/cmd.go

```diff
@@ -0,0 +1,157 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package org
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift-online/ocm-cli/pkg/arguments"
+	"github.com/openshift-online/ocm-cli/pkg/ocm"
+	table "github.com/openshift-online/ocm-cli/pkg/table"
+	amv1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+)
+
+var args struct {
+	columns   string
+	parameter []string
+	padding   int
+}
+
+// Cmd ...
+var Cmd = &cobra.Command{
+	Use:     "organization",
+	Aliases: []string{"organization", "org", "organizations"},
+	Short:   "List organizations.",
+	Long:    "Display a list of organizations.",
+	Args:    cobra.NoArgs,
+	RunE:    run,
+}
+
+func init() {
+	// Add flags to rootCmd:
+	fs := Cmd.Flags()
+	arguments.AddParameterFlag(fs, &args.parameter)
+	fs.StringVar(
+		&args.columns,
+		"columns",
+		"id,name", // Default value gets assigned later as connection is needed.
+		"Organization identifier. Defaults to the organization of the current user.",
+	)
+	fs.IntVar(
+		&args.padding,
+		"padding",
+		45,
+		"Takes padding for custom columns, default to 45.",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+
+	// Create the client for the OCM API:
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return fmt.Errorf("Failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Indices
+	pageIndex := 1
+	pageSize := 100
+
+	// Setting column names and padding size
+	// Update our column name displaying variable:
+	args.columns = strings.Replace(args.columns, " ", "", -1)
+	colUpper := strings.ToUpper(args.columns)
+	colUpper = strings.Replace(colUpper, ".", " ", -1)
+	columnNames := strings.Split(colUpper, ",")
+	paddingByColumn := []int{29, 65, 70}
+	if args.columns != "id,name" {
+		paddingByColumn = []int{args.padding}
+	}
+
+	// Print Header Row:
+	table.PrintPadded(os.Stdout, columnNames, paddingByColumn)
+
+	for {
+		// Next page request:
+		request := connection.AccountsMgmt().V1().Organizations().
+			List().
+			Page(pageIndex).
+			Size(pageSize)
+
+		// Apply parameters
+		arguments.ApplyParameterFlag(request, args.parameter)
+
+		// Fetch next page
+		orgList, err := request.Send()
+		if err != nil {
+			return fmt.Errorf("Failed to retrieve organization list: %v", err)
+		}
+
+		// Display organization information
+		orgList.Items().Each(func(org *amv1.Organization) bool {
+			// String to output marshal -
```

Comment:
This function interleaves 2 responsibilities: iterating over pages & printing an item accourding to args.columns.  Both are non of non-trivial size.  And both are nearly-repeated in multiple files.
Ideally, we should find ways to extract the logic for reuse.

But for now, can you simply move the printing code to a separate function here to split the responsibilities?

Reply:
[out of scope, just observation: it's funny we're using the SDK "typed" interface to deserialize response into structs, only to ser/deserialize again into interface{}.  I suspect it'd be _easier_ to extract generic page-walking and column-printing logics if we were using untyped `connect.Get()`.  
Maybe the best reuse unit could be a "half-typed" interface that did deserialize top-level `.Total`, `.Page`, etc but left `Items []interface{}`, which would allow reusing _a common result type for all collections_ (together with `Error` type which is already common).]

Reply:
[continuing my braindump: SDK could also use `json.RawMessage` to _delay the decision_ whether to unmarshal Items as typed or map[string]interface{} :gift:  https://eli.thegreenplace.net/2020/representing-json-structures-in-go/#hybrid-representation]